src/networking.c

```diff
@@ -1045,10 +1057,7 @@ void clientAcceptHandler(connection *conn) {
         DefaultUser->flags & USER_FLAG_NOPASS &&
         !(c->flags & CLIENT_UNIX_SOCKET))
```

Comment:
in theory we can now unify the two `if`s into one, but that will destroy the blame log for many lines.
however, let's at least remove the check for CLIENT_UNIX_SOCKET in the outer `if`.